internal/resources/cluster/data_source_cluster.go

```diff
@@ -129,18 +129,30 @@ func dataSourceTMCClusterRead(ctx context.Context, d *schema.ResourceData, m int
 
 	clusterSpec := constructSpec(d)
 
-	switch {
-	case resp.Cluster.Spec.TkgAws != nil:
-		resp.Cluster.Spec.TkgAws.Topology.NodePools = clusterSpec.TkgAws.Topology.NodePools
-	case resp.Cluster.Spec.TkgServiceVsphere != nil:
-		resp.Cluster.Spec.TkgServiceVsphere.Topology.NodePools = clusterSpec.TkgServiceVsphere.Topology.NodePools
-	case resp.Cluster.Spec.TkgVsphere != nil:
-		resp.Cluster.Spec.TkgVsphere.Topology.NodePools = clusterSpec.TkgVsphere.Topology.NodePools
-	}
+	getNodepoolForCluster(resp.Cluster.Spec, clusterSpec)
```

Comment:
Change the function name to setNodepoolForClusterResource(resp.Cluster.Spec, constructSpec(d))

and also call constructSpec(d) inside the parameter itself

Reply:
Done